KeshUI/Public/KeshUI/Container/KUIGridContainer.cpp

```diff
@@ -142,6 +142,28 @@ void UKUIGridContainer::AddColumn( float fWidth )
 }
 
 
+void UKUIGridContainer::RemoveColumn( uint8 iColumn )
+{
+	if (iColumn >= arColumns.Num( ))
+	{
+		KUIErrorUO( "Invalid column: %d", iColumn );
+		return;
+	}
+
+	arColumns.RemoveAt( iColumn );
+
+	for (int32 i = 0; i < arRows.Num( ); i++)
+	{
+		ar2Elements[ i ].SetNum( arColumns.Num( ) );
+		UKUIInterfaceElement *Child = ar2Elements [ i ] [ iColumn ].Get();
+		if ( Child != NULL )
+			RemoveChild ( Child );
+	}
```

Comment:
You need to use RemoveAt, not SetNum, to change element array or it won't move the column values down properly.

And you need to fetch the removed element before changing the array.